providers/implementations/exchange/ecx_exch.c.in

```diff
@@ -177,24 +180,32 @@ static void *ecx_dupctx(void *vecxctx)
     return dstctx;
 }
 
+#ifdef FIPS_MODULE
+{- produce_param_decoder('ecx_get_ctx_params',
+                         (['ALG_PARAM_FIPS_APPROVED_INDICATOR', 'ind', 'int'],
+                         )); -}
+#endif
+
 static const OSSL_PARAM *ecx_gettable_ctx_params(ossl_unused void *vctx,
                                                  ossl_unused void *provctx)
 {
-    static const OSSL_PARAM known_gettable_ctx_params[] = {
-        OSSL_FIPS_IND_GETTABLE_CTX_PARAM()
-        OSSL_PARAM_END
-    };
-    return known_gettable_ctx_params;
+#ifdef FIPS_MODULE
+    return ecx_get_ctx_params_list;
+#else
+    return NULL;
```

Comment:
Should we actually return `{ OSSL_PARAM_END }` here? Theoretically this change from returning something to returning `NULL` could be a surprise.

Reply:
Removing the conditionals matches what was there so I'll do that.